### PR TITLE
Tuchfarber/auth flow

### PR DESCRIPTION
**Pull request by tuchfarber**
Adds homepage, link to CMS for staff users, and login/logout links.

### CODE REVIEW

File: skateboard/apps/skateboard/templates/skateboard/home.html [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{% extends "base.html" %}
+
+{% block content %}
+    <div id="home" class="container">
+        <div class="row">
+            <div class="col-12">
+                <h2 class="center-text">Welcome to Skateboard!</h2>
+                <p class="center-text">Please login to view your skateboards</p>
```
Comment (bill-filler) [inline skateboard/apps/skateboard/templates/skateboard/home.html]: Let's make this message dynamic based on login. Currently it always says "Please login" even if you are logged in:
```
{% if user.is_authenticated %}
        <span class="welcome">Welcome, {{user.username}}</span>
        <a class="nav-link" href="/logout">Logout</a>
        {% else %}
        <a class="nav-link" href="/login">Login</a>
        {% endif %}
```

**Comment by bill-filler** [comment]
@tuchfarber 
login works great, just the one comment. Can we add redirection for logout so it goes back to the skateboard homepage after logout? Currently it stays on localhost:18000 after logout. I'm fine if you want to merge this and do that in a separate PR

**Comment by bill-filler** [comment]
Also based on our conversation yesterday, the user should be able to view the contents of the home page (PubIndexPage) and about page (PubHomePage) without being logged in, but the contents of the PubSubPage should require authentication AND authorization. I'm assuming we'll be able to do this once we have those pages in place.

File: skateboard/apps/skateboard/templates/skateboard/home.html (hunk 1 above)
Comment (tuchfarber) [reply skateboard/apps/skateboard/templates/skateboard/home.html]: Added.

**Comment by tuchfarber** [comment]
@bill-filler I don't think we can do anything about the logout flow. Since we're using the LMS's OAuth for login, I think we have to log out through there and stay there. I at least couldn't find a setting for it. 

In regards to the home/index/about pages, those are coming in a seperate PR. This was mainly to make development a little easier by standardizing the flow.

**Pull request merged**